runtime/core.go

```diff
@@ -1216,17 +1215,17 @@ func hashNotImplemented(f *Frame, o *Object) (*Object, *BaseException) {
 }
 
 // pyPrint encapsulates the logic of the Python print function.
-func pyPrint(f *Frame, args Args, sep, end string, file io.Writer) *BaseException {
+func pyPrint(f *Frame, args Args, sep, end string, file *File) *BaseException {
 	for i, arg := range args {
 		if i > 0 {
-			fmt.Fprint(file, sep)
+			file.writeString(sep)
```

Comment:
Would you mind adding the code to raise IOErrorType for these calls? The previous code should have been checking this. It'd be good to take care of this while we're in here.